core/config/README.md

```diff
@@ -0,0 +1,91 @@
+# Configuration Package
+
+At a high level, configuration is any data that is used in an application but
+not part of the application itself. Any reasonable complex system will need to
+have knobs to tune and not everything can have intelligent defaults.
+
+In UberFx, we try very hard to make sure that users only need to provide the
+minimal amount of configuration necessary to have working components, while also
+allowing them to override any field they need to if the defaults don't make
+sense for them.
+
+## Provider
```

Comment:
Would this be better as "ConfigurationProvider"?

Alternatively, you could have a "Providers" section with an H2 and then split it into two H3 parts: ConfigurationProvider and Dynamic Configuration Providers.

I'd like to see some distinction between this section and the next. Like if dynamic config providers are treated specially, maybe tell your reader up here that ConfigurationProvider is for static stuff too? Does that make sense?